src/services/permissions-checker.js

```diff
@@ -16,58 +16,144 @@ class PermissionsChecker {
+      this._setTeamsACLPermissions(permissions);
+    }
+  }
+
+  static getLastRetrieveTime(renderingId, permissionName) {
+    // In the case of rolesACL format and browseEnabled permission, the last retrieve to be taken
+    // into account is the one stored by rendering (because of the scope information).
+    if (PermissionsChecker.isrolesACLActivated && permissionName !== 'browseEnabled') {
+      return PermissionsChecker.permissions.collections.lastRetrieve;
+    }
+    return PermissionsChecker.permissions.renderings[renderingId]
+      ? PermissionsChecker.permissions.renderings[renderingId].lastRetrieve
+      : null;
   }
 
   static resetExpiration(renderingId) {
-    const permissions = PermissionsChecker.getPermissions(renderingId);
-    if (permissions) {
-      permissions.lastRetrieve = null;
+    if (PermissionsChecker.isrolesACLActivated && PermissionsChecker.permissions.collection) {
+      PermissionsChecker.permissions.collection.lastRetrieve = null;
+    }
+
+    if (PermissionsChecker.permissions.renderings[renderingId]) {
+      PermissionsChecker.permissions.renderings[renderingId].lastRetrieve = null;
     }
   }
 
+  static _isPermissionAllowed(permissionValue, userId) {
+    return Array.isArray(permissionValue)
+      ? permissionValue.includes(parseInt(userId, 10))
+      : permissionValue;
+  }
+
   static _isSmartActionAllowed(smartActionsPermissions, permissionInfos) {
     if (!permissionInfos
       || !permissionInfos.userId
-      || !permissionInfos.actionId
+      || !permissionInfos.actionName
       || !smartActionsPermissions
-      || !smartActionsPermissions[permissionInfos.actionId]) {
+      || !smartActionsPermissions[permissionInfos.actionName]) {
       return false;
     }
 
-    const { userId, actionId } = permissionInfos;
-    const { allowed, users } = smartActionsPermissions[actionId];
+    const { userId, actionName } = permissionInfos;
+    const { triggerEnabled } = smartActionsPermissions[actionName];
 
-    return allowed && (!users || users.includes(parseInt(userId, 10)));
+    return PermissionsChecker._isPermissionAllowed(triggerEnabled, userId);
   }
 
   // NOTICE: Compute a scope to replace $currentUser variables with
   //         the actual user values. This will generate the expected
   //         conditions filters when applied on the server scope response
-  static _computeConditionFiltersFromScope(userId, collectionListScope) {
-    const computedConditionFilters = _.clone(collectionListScope.filter);
+  static _computeConditionFiltersFromScope(userId, scope) {
+    const computedConditionFilters = _.clone(scope.filter);
     computedConditionFilters.conditions.forEach((condition) => {
       if (condition.value
         && `${condition.value}`.startsWith('$')
-        && collectionListScope.dynamicScopesValues.users[userId]) {
-        condition.value = collectionListScope
+        && scope.dynamicScopesValues.users[userId]) {
+        condition.value = scope
```

Comment:
This looks like a refacto and should not be part of this PR

Reply:
Yes sorry, hopefully you don't mind!